src/openrct2/ride/Vehicle.cpp

```diff
@@ -9457,6 +9480,118 @@ static sint32 vehicle_update_track_motion_mini_golf(rct_vehicle * vehicle, sint3
     return regs.eax;
 }
 
+/**
+ *
+ *  rct2: 0x006DC1E4
+ */
+static void vehicle_update_track_motion_powered_ride_acceleration(rct_vehicle * vehicle, rct_ride_entry_vehicle * vehicleEntry, uint32 totalMass, sint32 * acceleration)
+{
+    if (vehicleEntry->flags & VEHICLE_ENTRY_FLAG_POWERED_RIDE_UNRESTRICTED_GRAVITY)
+    {
+        if (vehicle->velocity > (vehicle->speed * 0x4000))
+        {
+            // Same code as none powered rides
+            if (*acceleration <= 0)
+            {
+                if (*acceleration >= -500)
+                {
+                    if (vehicle->velocity <= 0x8000)
```

Comment:
Could we collapse these nested conditionals into one block?

Reply:
Probably. I was planning on breaking this code out into a separate function as it's used in 3+ places